contrib/infiniband/ibvctx.c

```diff
@@ -1116,16 +1116,26 @@ int _get_cq_event(struct ibv_comp_channel * channel,
   flags = fcntl(internal_channel->real_channel->fd, F_GETFL, NULL);
 
   // We need to change the call to non-blocking mode
-  if (flags & O_NONBLOCK == 0) {
+  // FIXME: we need to take care of the case where fd is already
+  // changed to non-blocking mode by the user.
+  if (!(flags & O_NONBLOCK)) {
+    rslt = fcntl(internal_channel->real_channel->fd,
+                 F_SETFL, flags | O_NONBLOCK);
+    if (rslt < 0) {
+      fprintf(stderr, "Failed to change file descriptor "
```

Comment:
The internal print is mainly used to print some debug info. Here it indicates an error.